### DISCUSSION TITLE
Replaying converted ROS bag file (.mcap) in ROS2 environment

### DISCUSSION THREAD

**Discussion by holtvogt** [foxglove/community#747] [General]
Using the `MCAP` CLI, I managed to convert a `.bag` into `.mcap` format with

```bash
mcap convert <bag> <mcap>
```

However, within the topics the `Serialization Format` is `ros1` using the `ros2 bag info <mcap>` command. Apparently, this is an issue for `ros2 bag play <mcap>` as it shows the following error:

```txt
[INFO] [...] [rosbag2_cpp]: No plugin found providing serialization format 'ros1'. Falling back to checking RMW implementations.
[ERROR] [...] [rosbag2_cpp]: Could not initialize RMWImplementedConverter: No RMW implementation found supporting serialization format ros1
```

Previously, I installed the necessary packages with

```bash
sudo apt-get install ros-$ROS_DISTRO-ros2bag ros-$ROS_DISTRO-rosbag2*
```

, whereas I am currently running an Ubuntu 22.04 with ROS Humble.

**Comment by defunctzombie** [comment]
It is not possible to play a ROS 1 .bag file in a ROS 2 environment. ROS 2 uses a completely different serialization format for messages which is what the error is telling you - there is no format support for "ros1".

**Comment by holtvogt** [comment]
The file played on ROS2 is the converted `.mcap` one. That should be compatible with ROS2, shouldn't it?

**Reply by defunctzombie** [reply]
MCAP is a container format that can store many different kinds of messages (protobuf, ros1, ros2 (cdr), flatbuffer, json, etc). Whether you can actually "play" or "read" the data depends on the framework or system you want to use the mcap file with. In the case of ROS 2, this means supporting a specific kind of message encoding. So while the `ros2 bag play` code was able to open the mcap file (since you have a valid mcap file), when the player went to actually read the message data and feed it into a ROS 2 system it encountered messages that are not supported for a ROS 2 system.
